### PR TITLE
Fix bug where DoubleMLLPLR was normalising by outer folds

### LINKED ISSUE #401: [Bug]: DoubleMLLPLR coefficient is incorrect when n_folds ≠ n_folds_inner, due to normalisation error
### Describe the bug

The DoubleMLLPLR model estimation implicitly assumes that `n_folds == n_folds_inner`. When the two are not equal, the normalisation used to get the coefficient is no longer correct. 

I believe it stems from here: https://github.com/DoubleML/doubleml-for-py/blob/c1f0f555efa072446944313995bd493e1de658d3/doubleml/utils/_estimation.py#L220-L231

I would be happy to open a PR to patch this.

### Minimum reproducible code snippet

```python
import numpy as np
import doubleml as dml
from doubleml.plm.datasets import make_lplr_LZZ2020
from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor

SEED = 123
np.random.seed(SEED)

dml_data = make_lplr_LZZ2020(alpha=0.5, n_obs=2000, dim_x=20)

ml_M = RandomForestClassifier(random_state =SEED)
ml_t = RandomForestRegressor(random_state =SEED)
ml_m = RandomForestRegressor(random_state =SEED)

# Fit with n_folds == n_folds_inner (OK)
dml_lplr = dml.DoubleMLLPLR(dml_data, ml_M, ml_t, ml_m, n_folds=5, n_folds_inner=5)
dml_lplr.fit()
print(dml_lplr.summary)

# shows:
#       coef   std err         t     P>|t|    2.5 %    97.5 %
# d  0.466365  0.071741  6.500725  7.993379e-11  0.325757  0.606974

# Fit with n_folds < n_folds_inner (biased upwards)
dml_lplr2 = dml.DoubleMLLPLR(dml_data, ml_M, ml_t, ml_m, n_folds=3, n_folds_inner=10)
dml_lplr2.fit()
print(dml_lplr2.summary)

# shows:
#       coef   std err         t     P>|t|    2.5 %    97.5 %
# d  0.869113  0.07858  11.060204  1.956524e-28  0.715099  1.023128
```

### Expected Result

Would expect to encounter an assertion error or to have the normalisation procedure adjust appropriately when `n_folds != n_folds_inner`

### Actual Result

No error/warning is shown but the estimation is incorrect when `n_folds != n_folds_inner`

### Versions

Windows-11
Python 3.13.1
DoubleML 0.11.4
Scikit-Learn 1.9.0

### PR DESCRIPTION
**Pull request by masonrhayes**
### Description

This PR changes a few lines of code so that the `_double_dml_cv_predict()` function correctly normalises by inner folds. 

### Reference to Issues or PRs
Fixes #401, where the DoubleMLLPLR model would produce incorrect coefficient estimates when `n_folds != n_folds_inner`. 

### PR Checklist
Please fill out this PR checklist (see our [contributing guidelines](https://github.com/DoubleML/doubleml-for-py/blob/main/CONTRIBUTING.md#checklist-for-pull-requests-pr) for details).

- [x] The title of the pull request summarizes the changes made.
- [x] The PR contains a detailed description of all changes and additions.
- [x] References to related issues or PRs are added.
- [x] The code passes all (unit) tests - I have not run the full test suite but only targeted tests for the LPLR model. The function `_double_dml_cv_predict()` is only used in `plm/lplr.py` so should not have any effects on other functionality.
- [ ] Enhancements or new feature are equipped with unit tests. **(N/A)** - bug fix only.
- [x] The changes adhere to the PEP8 standards.

### CODE REVIEW

**Comment by SvenKlaassen** [comment]
Thank you. I just added a small test.

**Review by SvenKlaassen — APPROVED** [review]
